ocaml/typing/typeclass.ml

```diff
@@ -1175,14 +1175,20 @@ and class_expr_aux cl_num val_env met_env virt self_scope scl =
           default;
        ]
       in
+      let param_suffix =
+        match l with
+        | Nolabel -> ""
+        | Labelled name | Optional name -> name
```

Comment:
I think you could either `assert false` or some other error in the non-optional cases (see assertion in `typecore.ml`, line 4370)